src/Tribe/Log/File_Logger.php

```diff
@@ -0,0 +1,265 @@
+	}
+
+	/**
+	 * Sets the path for the log file we're currently interested in using.
+	 *
+	 * @param string $date = null
+	 */
+	protected function set_log_file( $date = null ) {
+		$this->log_file = $this->get_log_file_name( $date );
+		$this->obtain_handle();
+	}
+
+	/**
+	 * Used to switch between contexts for reading ('r') and writing
+	 * ('a' := append) modes.
+	 *
+	 * @see fopen() documentation
+	 *
+	 * @param string $context
+	 */
+	protected function set_context( $context ) {
+		$this->context = $context;
+		$this->close_handle();
+		$this->obtain_handle();
+	}
+
+	/**
+	 * Attempts to obtain a file handle for the current log file.
+	 */
+	protected function obtain_handle() {
+		$this->close_handle();
+		$this->handle = fopen( $this->log_file, $this->context );
+	}
+
+	/**
+	 * Closes the current file handle, if one is open.
+	 */
+	protected function close_handle() {
+		// is_resource() only returns true for open resources
+		if ( is_resource( $this->handle ) ) {
+			fclose( $this->handle );
+		}
+	}
+
+	/**
+	 * Returns the log name to be used for reading/writing events for a specified date
+	 * (defaulting to today, if no date is specified).
+	 *
+	 * @param string $date = null
+	 *
+	 * @return string
+	 */
+	protected function get_log_file_name( $date = null ) {
+		if ( null === $date ) {
+			$date = date_i18n( 'Y-m-d' );
+		}
+
+		$filename = $this->log_dir . DIRECTORY_SEPARATOR . $this->get_log_file_basename() . $date . '.log';
+
+		/**
+		 * Dictates the filename of the log used to record events for the specified date.
+		 *
+		 * @var string $filename
+		 * @var string $date
+		 */
+		return apply_filters( 'tribe_file_logger_filename', $filename, $date );
+	}
+
+	protected function get_log_file_basename() {
+		/**
+		 * Log files share a common prefix, which aids identifying archived/rotated logs.
+		 * This filter allows a degree of control to be exercised over the prefix to avoid
+		 * conflicts, etc.
+		 *
+		 * @var string $log_file_base_name
+		 */
+		return apply_filters( 'tribe_file_logger_file_prefix', $this->module_id . '_' );
+	}
+
+	/**
+	 * Returns a 'human friendly' name for the logging implementation.
+	 *
+	 * @return string
+	 */
+	public function get_name() {
+		return __( 'Default (uses temporary files)', 'tribe-common' );
+	}
+
+	/**
+	 * Indicates if the logger will work in the current environment.
+	 *
+	 * @return bool
+	 */
+	public function is_available() {
+		return is_writable( $this->log_dir ) && is_readable( $this->log_dir );
+	}
+
+	/**
+	 * Responsible for commiting the entry to the log.
+	 *
+	 * @param string $entry
+	 * @param string $type
+	 * @param string $src
+	 */
+	public function log( $entry, $type = Tribe__Log::DEBUG, $src = '' ) {
+		// Ensure we're in 'append' mode before we try to write
+		if ( 'a' !== $this->context ) {
+			$this->set_context( 'a' );
+		}
+
+		fputcsv( $this->handle, array( date_i18n( 'Y-m-d H:i:s' ), $entry, $type, $src ) );
```

Comment:
A lot of these methods would fail on shared hosting erratically.
We should rely on the Filesystem API if we really want to write any file at all. Both in reading, writing and existence operations.